python/dpu_utils/utils/richpath.py

```diff
@@ -343,44 +389,49 @@ def get_size(self) -> int:
     def exists(self) -> bool:
         return os.path.exists(self.path)
 
+    def delete(self, missing_ok: bool=True) -> None:
+        try:
+            os.unlink(self.path)
+        except FileNotFoundError:
+            if not missing_ok:
+                raise
+
     def to_local_path(self) -> 'LocalPath':
         return self
 
     def _copy_from_local_file(self, local_file: 'LocalPath') -> None:
-        os.makedirs(os.path.dirname(self.path) ,exist_ok=True)
+        os.makedirs(os.path.dirname(self.path), exist_ok=True)
         shutil.copy2(src=local_file.path, dst=self.path)
 
 
 class AzurePath(RichPath):
-    def __init__(self, path: str, azure_container_name: str, azure_blob_service: BlockBlobService,
+    def __init__(self, path: str, azure_container_client: ContainerClient,
                  cache_location: Optional[str]):
         super().__init__(path)
-        self.__container_name = azure_container_name
-        self.__blob_service = azure_blob_service
+        self.__container_client = azure_container_client
+        self.__blob_client = self.__container_client.get_blob_client(self.path)  # TODO: Does this work with dirs?

```

Comment:
Did you get around to test the TODO?

Reply:
Apparently it works and it's tested [here](https://github.com/microsoft/dpu-utils/blob/1c6478caf6db7ec19640fcb95dcf384baad1ba05/python/tests/utils/test_richpath.py#L160). Thanks for pointing out the missing TODO.
```suggestion
        self.__blob_client = self.__container_client.get_blob_client(self.path)

```